packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -35,3 +60,424 @@ def __init__(self, **kwargs):
         :param performative: the type of message.
         """
         super().__init__(**kwargs)
+        self.w3 = Web3(Web3.HTTPProvider(PROVIDER))
```

Comment:
this should not be instantiated here, instead, I would pass in the relevant ledger_api as an argument of the function which needs it.